config/env/env_types.go

```diff
@@ -0,0 +1,161 @@
+package env
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var TypeError = errors.New("type could not be converted")
+var EnvironmentError = errors.New("environment variable not found")
+
+var regex *regexp.Regexp
+
+func init() { regex = regexp.MustCompile(`\${[A-Z]+[A-Z1-9_]*}`) }
+
+func replaceEnvVar(in string) (string, error) {
+
+	// TODO(ear7h): FindAllString
+	varName := regex.FindString(in)
```

Comment:
You should be getting all the possible replacements; here. This would be useful in a bind like config option, where the user might put something like: `${PG_HOST}:${PG_PORT}`

I would use the [FindAllStringIndex](https://godoc.org/regexp#Regexp.FindAllStringIndex) function. That way you know the index of the places you need to substitute each of the parts.